spec/controllers/users/backup_code_setup_controller_spec.rb

```diff
@@ -8,7 +8,8 @@
         :authenticate_user!,
         :confirm_user_authenticated_for_2fa_setup,
         :apply_secure_headers_override,
-        [:confirm_recently_authenticated_2fa, except: ['reminder']],
+        [:confirm_recently_authenticated_2fa, except: ['reminder', 'continue']],
+        :validate_internal_referrer?,
```

Comment:
By test coverage, I was hoping to see something which validates that the user would be redirected if they attempt to visit the page without satisfying the preconditions (and vice-versa, not redirected if they do satisfy the preconditions). We don't have any test coverage for that.